src/main/java/org/tikv/common/operation/RegionErrorHandler.java

```diff
@@ -264,8 +269,23 @@ public TiRegion getRegion() {
     return recv.getRegion();
   }
 
-  private void invalidateRegionStoreCache(TiRegion ctxRegion) {
+  private void invalidateRegionStoreCache(TiRegion ctxRegion, long storeId) {
     regionManager.invalidateRegion(ctxRegion);
-    regionManager.invalidateStore(ctxRegion.getLeader().getStoreId());
+    regionManager.invalidateStore(storeId);
+    notifyRegionStoreCacheInvalidate(
+        ctxRegion.getId(), storeId, CacheInvalidateEvent.CacheType.REGION_STORE);
+  }
+
+  private void notifyRegionStoreCacheInvalidate(
+      long regionId, long storeId, CacheInvalidateEvent.CacheType type) {
+    if (cacheInvalidateCallBack != null) {
+      cacheInvalidateCallBack.apply(new CacheInvalidateEvent(regionId, storeId, true, true, type));
```

Comment:
should catch error, callback must not break the main routine.
Also, it should be executed synchronizedly

Reply:
Already add